maps/src/main/java/org/odk/collect/maps/layers/OfflineMapLayersViewModel.kt

```diff
@@ -91,9 +91,12 @@ class OfflineMapLayersViewModel(
         settingsProvider.getUnprotectedSettings().save(ProjectKeys.KEY_REFERENCE_LAYER, layerId)
     }
 
-    fun deleteLayer(layerId: String) {
+    fun deleteLayer(layerId: String, checkedLayerDeleted: Boolean) {
         _isLoading.value = true
         scheduler.immediate {
+            if (checkedLayerDeleted) {
```

Comment:
No, I need to pass either this boolean or `checkedLayerId` and then compare `layerId` with that `checkedLayerId` to know if this is the selected layer as this view model has no knowledge about the state.

Reply:
We can grab it from settings right? The tests pass with the following:

```kotlin
if (layerId == settingsProvider.getUnprotectedSettings().getString(ProjectKeys.KEY_REFERENCE_LAYER)) {
    saveCheckedLayer(null)
}

Reply:
Actually there are 3 cases:
1. You can delete the layer that is saved and selected at the same time
2. You can delete the layer that is selected but not saved.
3. You can delete the layer that is saved but not selected.

Do you agree to reset to none in all those cases for simplicity?

Reply:
This got really stuck in my head, so I wanted to quickly sketch out how a generic select view model might handle this:

```kotlin
data class SelectItem(val id: String, val selected: Boolean)
class SelectViewModel(data: LiveData<List<SelectItem>>) {

    private val _selected = MutableLiveData<String?>()
    val selected: LiveData<String?> = LiveDataUtils.zip(_selected, data).map { (selected, items) ->
        if (selected != null && items.any { it.id == selected }) {
            selected
        } else {
            items.find { it.selected }?.id
        }
    }

    fun select(id: String) {
        _selected.value = id
    }
}
```

Here'd you'd `map` the underlying data (existing layers and settings) into a `LiveData<SelectItem>`. The `zip` and `map` then allows you to combine that with a mutable selection, and you can basically say: "if I've selected one and it's in the data use that and if not use the selected one from the data (if there is one)".

Reply:
I've improved the fix with the first approach please take a look.
If you strongly feel like we should have a more generic approach here feel free to push your changes because I think I might not have enough time to understand your way of thinking and implement those changes before my vacation.

Reply:
Yeah, that's fine. To be clear, I agree the generic approach is better as a follow-up, but I needed to get that first thought out of my head 😆. I can always give that a go next week so we can discuss when you're back!

Reply:
If you want to keep it open then https://github.com/getodk/collect/pull/6188 shouldn't be blocked I think. https://github.com/getodk/collect/pull/6188 brings more changes so if one of them needs to be blocked it should be rather this one.